### PR TITLE
State: Add getJetpackOnboardingProgress selector

### PR DESCRIPTION
**Pull request by tyxla**
This PR introduces a `getJetpackOnboardingProgress` selector, which is essentially a port of https://github.com/Automattic/wp-calypso/blob/master/client/jetpack-onboarding/steps/summary.jsx#L109

To test:

* Checkout this branch
* Verify all tests pass:

```
npm run test-client client/state/selectors/test/get-jetpack-onboarding-progress.js
```

### CODE REVIEW

File: client/state/selectors/get-jetpack-onboarding-progress.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+/**
+ * Internal dependencies
+ */
+import { isJetpackOnboardingStepCompleted } from 'state/selectors';
+import { JETPACK_ONBOARDING_STEPS as STEPS } from 'jetpack-onboarding/constants';
+
+export default function getJetpackOnboardingProgress( state, siteId, steps ) {
+	if ( ! steps ) {
+		steps = values( STEPS );
+	}
```
Comment (ockham) [inline client/state/selectors/get-jetpack-onboarding-progress.js]: Hmm, not a big fan of this sort of fallbacks. I find they tend to make things less predictable. Can we either drop this, or fully embrace it (i.e. drop the `steps` arg to the selector)?

File: client/state/selectors/test/get-jetpack-onboarding-progress.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	const state = {
+		jetpackOnboarding: {
+			settings: {
+				[ siteId ]: {},
+			},
+		},
+	};
+
+	test( 'should return progress of all steps if no steps are specified', () => {
+		const steps = invert( STEPS );
```
Comment (ockham) [inline client/state/selectors/test/get-jetpack-onboarding-progress.js]: That seems a tad _too_ arbitrary, even for a test 😄

File: client/state/selectors/test/get-jetpack-onboarding-progress.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+	test( 'should return onboading progress for the specified steps', () => {
+		const siteId = 2916284;
+		const state = {
+			jetpackOnboarding: {
+				settings: {
+					[ siteId ]: {},
+				},
+			},
+		};
+		const steps = [ [ STEPS.SITE_TITLE ], [ STEPS.SITE_TYPE ] ];
```
Comment (ockham) [inline client/state/selectors/test/get-jetpack-onboarding-progress.js]: Working on #21565, I just discovered that the brackets around `STEPS.SITE_TITLE  ` and `STEPS.SITE_TYPE` here are probably wrong/unneeded: They're array items, not object keys, so they will be interpreted as variable names, as they should.

Comment (tyxla) [reply client/state/selectors/test/get-jetpack-onboarding-progress.js]: 🤦‍♂️ 

PR coming in a bit.

Comment (tyxla) [reply client/state/selectors/test/get-jetpack-onboarding-progress.js]: #21593, thanks Bernie

**Pull request merged**